Helper/DeviceView.php

```diff
@@ -54,14 +54,12 @@ class DeviceView
      */
     public function __construct(Container $serviceContainer)
     {
-        if (false === $serviceContainer->isScopeActive('request')) {
+        if (!$this->request = $serviceContainer->get('request_stack')->getMasterRequest()) {
```

Comment:
Well put, a and is precisely why I made the suggestion.

> On 9 Feb 2016, at 23:12, netmikey <notifications@github.com> wrote:
> 
> In Helper/DeviceView.php:
> 
> > @@ -54,14 +54,12 @@ class DeviceView
> >       */
> >      public function __construct(Container $serviceContainer)
> >      {
> > -        if (false === $serviceContainer->isScopeActive('request')) {
> > +        if (!$this->request = $serviceContainer->get('request_stack')->getMasterRequest()) {
> Bumping the major version doesn't mean the (then) older major won't work on Symfony 2.3 anymore. If need be, we could even create a branch and release a patch on the (then) older major.
> I've seen a couple of bundles taking this approach to avoid the boilerplate code required to support both sf <= 2.3 and sf >= 2.4 at the same time. Plus, once sf 2.3 goes end-of-life, we'd have to go through our codebase (again) and explicitly remove the boilerplate code, which takes time (again).
> So personally, I like the idea of bumping the major version and clearly documenting the intention behind it in the docs and the composer.json's dependency declarations.
> 
> —
> Reply to this email directly or view it on GitHub.
>